modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/GeneratedClass.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.execution;
+
+import java.io.File;
+
+/**
+ * Represents a class that was dynamically generated at runtime
+ *
+ * @param <T> the generic type of the generated class
+ * @since 4.3.0
+ */
+public final class GeneratedClass<T> {
+
+  private final Class<T> generatedClass;
+  private final File byteCodeFile;
```

Comment:
the file should only be generated if a certain troubleshooting flag is enabled

Reply:
The underlying classloader is an URLClassLoader. How can you load the class otherwise?

Reply:
good point